tests/src/e2e-smoke.spec.ts

```diff
@@ -1,51 +1,48 @@
+/**********************************************************************
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
 import type { BrowserWindow } from 'electron';
-import type { ElectronApplication, JSHandle, Page } from 'playwright';
-import { _electron as electron } from 'playwright';
+import type { JSHandle, Page } from 'playwright';
 import { afterAll, beforeAll, expect, test, describe } from 'vitest';
 import { expect as playExpect } from '@playwright/test';
-import { existsSync } from 'node:fs';
-import { rm } from 'node:fs/promises';
+import { PodmanDesktopRunner } from './runner/podman-desktop-runner';
+import { WelcomePage } from './model/pages/welcome-page';
+import { DashboardPage } from './model/pages/dashboard-page';
 
 const navBarItems = ['Dashboard', 'Containers', 'Images', 'Pods', 'Volumes', 'Settings'];
-let electronApp: ElectronApplication;
+let pdRunner: PodmanDesktopRunner;
 let page: Page;
 
 beforeAll(async () => {
-  // remove all videos/screenshots
-  if (existsSync('tests/output')) {
-    console.log('Cleaning up output folder...');
-    await rm('tests/output', { recursive: true, force: true });
-  }
-
-  const env: { [key: string]: string } = Object.assign({}, process.env as { [key: string]: string });
-  env.PODMAN_DESKTOP_HOME_DIR = 'tests/output/podman-desktop';
-
-  electronApp = await electron.launch({
-    args: ['.'],
-    env,
-    recordVideo: {
-      dir: 'tests/output/videos',
-      size: {
-        width: 1050,
-        height: 700,
-      },
-    },
-  });
-
-  page = await electronApp.firstWindow();
+  pdRunner = new PodmanDesktopRunner();
+  page = await pdRunner.start();
 });
 
 afterAll(async () => {
-  await electronApp.close();
+  await pdRunner.close();
 });
 
 describe('Basic e2e verification of podman desktop start', async () => {
   describe('Welcome page handling', async () => {
     test('Check the Welcome page is displayed', async () => {
-      // Direct Electron console to Node terminal.
-      page.on('console', console.log);
+      pdRunner.redirectEletronConsoleToTerminal();
```

Comment:
there is typo
```suggestion
      pdRunner.redirectElecronConsoleToTerminal();
```

but could it also be done automatically by the runner ?

Reply:
I mean, by default, it redirects the logs, no need to ask it on each test